.travis.yml

```diff
@@ -26,23 +26,20 @@ before_install:
 
 install:
 - npm install
-- if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then cd darwin && npm i && cd ..; fi
 - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then cd linux && npm i && cd ..; fi
+- if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then cd darwin && npm i && cd ..; fi
 
 before_script:
-- if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
-    export DISPLAY=:99.0;
-    sh -e /etc/init.d/xvfb start;
-    sleep 3;
-    npm run lint linux;
-    npm run lint shared;
-    npm run lint test;
-    cd linux;
-  fi
-- if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
-    npm run lint darwin;
-    cd darwin;
-  fi
+- if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then export DISPLAY=:99.0; fi
+- if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then sh -e /etc/init.d/xvfb start; fi
+- if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then sleep 3; fi
+- if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then npm run lint linux; fi
+- if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then npm run lint shared; fi
+- if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then npm run lint test; fi
+- if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then cd linux; fi
```

Comment:
can we create a script in an external file with a `set -e` in it? this seems overly robust.

Reply:
We can but not for all commands on Travis. The script will have only these lines:
```bash
if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
  npm run lint linux;
  npm run lint shared;
  npm run lint test;
fi

if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
  npm run lint darwin;
fi
```
Which right now are only 4 lines from the whole `.travis.yml`. We still need to call the `export DISPLAY` and run `xvfb` directly from `.travis.yml` and also do all the `cd linux` or `cd darwin` directly from `.travis.yml`, all those on their own line.
I thought that creating a new script file just for 4 lines was not worth it. Let me know what you prefer